kpi/models/asset.py

```diff
@@ -859,6 +861,11 @@ def save(
     ):
         is_new = self.pk is None
 
+        if self.owner:
+            if is_new:
+                self.created_by = self.owner.username
+            self.last_modified_by = self.owner.username
+
```

Comment:
You better to do it at the API level. 
The asset could be modified by someone else with correct permissions.